src/content/docs/apm/agents/ruby-agent/api-guides/ruby-ai-monitoring-apis.mdx

```diff
@@ -0,0 +1,99 @@
+---
+title: AI monitoring APIs
+tags:
+  - Agents
+  - Ruby agent
+  - API guides
+metaDescription: 'For information about customizing New Relic''s Ruby agent for AI monitoring.'
+freshnessValidatedDate: never
+---
+
+When you've instrumented your app for AI monitoring, the New Relic Ruby agent automatically collects many AI metrics, but also provides APIs for collecting information on token count and user feedback.
+
+<Callout variant="tip">
+  AI monitoring APIs are available in Ruby agent version 9.8.0 and higher.
+</Callout>
+
+## Token count [#token-count]
+
+You can set a callback proc for calculating `token_count` attributes for LlmEmbedding and LlmChatCompletionMessage events, and then pass that information to New Relic using the `NewRelic::Agent.set_llm_token_count_callback` API.
+
+This API should be called only once to set a callback for use with all LLM token calculations. If it is called multiple times, each new callback will replace the old one. The proc will be called with a single hash as its input argument and must return an Integer representing the number of tokens used for that particular prompt, completion message, or embedding. Values less than or equal to 0 will not be attached to an event.
+
+The hash has the following keys:
+
+    * `:model` (String) - The name of the LLM model
+    * `:content` (String) - The message content or prompt
+
+The following example code demonstrates setting a callback that calculates token count and passing that callback to `NewRelic::Agent.set_llm_token_count_callback`.
+
+```rb
+    require 'tiktoken_ruby' # Example library for counting GPT model tokens used
+
+    token_count_callback = proc do |hash|
+    return unless hash[:model].includes?('gpt')
+
+    enc = Tiktoken.encoding_for_model(hash[:model])
+    enc.encode(hash[:content]).length
+    end
+
+    NewRelic::Agent.set_llm_token_count_callback(token_count_callback)
+```
```

Comment:
@akristen we need to add some indentation here for proper ruby code ◡̈ 
```suggestion
```rb
    require 'tiktoken_ruby' # Example library for counting GPT model tokens used

    token_count_callback = proc do |hash|
      return unless hash[:model].includes?('gpt')
  
      enc = Tiktoken.encoding_for_model(hash[:model])
      enc.encode(hash[:content]).length
    end

    NewRelic::Agent.set_llm_token_count_callback(token_count_callback)
```
```